### PR TITLE
feat: Adjustments to spec and enhanced event validation.

### PR DESCRIPTION
**Pull request by kinyoklion**

### CODE REVIEW

File: packages/shared/sdk-server/__tests__/MigratioOpEvent.test.ts [hunk 1]
```diff
@@ -144,7 +144,7 @@ describe('given an LDClient with test data', () => {
           expect(events.length).toBe(2);
           // Only check the measurements component of the event.
           const migrationEvent = events[1] as InputMigrationEvent;
-          expect(migrationEvent.measurements[0].key).toEqual('latency');
+          expect(migrationEvent.measurements[0].key).toEqual('latency_ms');
```
Comment (kinyoklion) [inline packages/shared/sdk-server/__tests__/MigratioOpEvent.test.ts:147]: New name, has units. I <3 units.

File: packages/shared/sdk-server/src/MigrationOpEventConversion.ts [hunk 2]
```diff
@@ -1,19 +1,214 @@
 import { internal, TypeValidators } from '@launchdarkly/js-sdk-common';
```
Comment (kinyoklion) [inline packages/shared/sdk-server/src/MigrationOpEventConversion.ts:1]: A migration event can come from external to the SDK. So this ensures we only put things we recognize into events and also that they vaguely look like events.

File: packages/shared/sdk-server/src/api/data/LDMigrationOpEvent.ts [hunk 3]
```diff
@@ -1,35 +1,60 @@
...
-        old?: number;
-        new?: number;
-      };
-    }
-  | {
-      key: 'consistent';
-      value: number;
-      samplingOdds: number;
-    };
+  | LDMigrationLatencyMeasurement
```
Comment (kinyoklion) [inline packages/shared/sdk-server/src/api/data/LDMigrationOpEvent.ts:54]: Added custom and made these easier to discriminate for validation.

File: packages/shared/sdk-server/__tests__/MigratioOpEvent.test.ts [hunk 4]
```diff
@@ -379,3 +381,98 @@ describe('given an LDClient with test data', () => {
...
+        kind: 'custom',
+        values: {},
+      },
+    ],
+  };
+  const validatedEvent = MigrationOpEventConversion(inputEvent);
+  expect(validatedEvent).toEqual(inputEvent);
+});
+
+it('removes bad custom measurements', () => {
```
Comment (kinyoklion) [inline packages/shared/sdk-server/__tests__/MigratioOpEvent.test.ts:436]: I didn't validate all the bad cases. The positive cases are covered by the individual migrations.

**Review by yusinto — APPROVED** [review]

**Review by keelerm84 — APPROVED** [review]

**Pull request merged**